src/execution/compiler/operator/analyze_translator.cpp

```diff
@@ -0,0 +1,405 @@
+  pg_statistic_column_lookup_[catalog::postgres::PgStatistic::STA_NUMROWS.oid_] = num_rows_;
+
+  for (const auto &col_info : catalog::postgres::PgStatistic::ANALYZE_AGGREGATES) {
+    auto agg_name = col_info.name_;
+    auto agg_var = codegen->MakeFreshIdentifier(std::string(agg_name));
+    aggregate_variables_.emplace_back(agg_var);
+    pg_statistic_column_lookup_[col_info.column_oid_] = agg_var;
+  }
+}
+
+void AnalyzeTranslator::PerformPipelineWork(WorkContext *context, FunctionBuilder *function) const {
+  const auto &plan = GetPlanAs<planner::AnalyzePlanNode>();
+  auto *codegen = GetCodeGen();
+
+  NOISEPAGE_ASSERT(plan.GetChild(0)->GetOutputSchema()->NumColumns() ==
+                       catalog::postgres::PgStatistic::NUM_ANALYZE_AGGREGATES * plan.GetColumnOids().size() + 1,
+                   fmt::format("We should 1 aggregate for the number of rows and then {} aggregates per column",
+                               catalog::postgres::PgStatistic::NUM_ANALYZE_AGGREGATES)
+                       .c_str());
+
+  // var pg_statistic_col_oids: [num_cols]uint32
+  // pg_statistic_col_oids[i] = ...
+  SetPgStatisticColOids(function);
+
+  // Declare/Initialize variables to insert into pg_statistic
+  InitPgStatisticVariables(context, function);
+
+  // var pg_statistic_index_iterator : IndexIterator
+  DeclarePgStatisticIterator(function);
+
+  // var pg_statistic_index_pr : *ProjectedRow
+  DeclarePgStatisticIndexPR(function);
+
+  // Perform a lookup and update for each column
+  for (size_t column_offset = 0; column_offset < plan.GetColumnOids().size(); column_offset++) {
+    // Assign all the aggregate variables values corresponding to the current column
+    AssignColumnStatistics(context, function, column_offset);
+
+    // @indexIteratorInit(&pg_statistic_index_iterator, queryState.execCtx, num_attrs, table_oid, index_oid,
+    // pg_statistic_col_oids)
+    InitPgStatisticIterator(function);
+
+    // pg_statistic_index_pr = @indexIteratorGetPR(&pg_statistic_index_iterator)
+    // @prSet(pr, type, nullable, attr, expr, false)
+    InitPgStatisticIndexPR(function);
+
+    // @indexIteratorScanKey(&pg_statistic_index_iterator)
+    auto *scan_call =
+        codegen->CallBuiltin(ast::Builtin::IndexIteratorScanKey, {codegen->AddressOf(pg_statistic_index_iterator_)});
+    auto *loop_init = codegen->MakeStmt(scan_call);
+    // @indexIteratorAdvance(&pg_statistic_index_iterator)
+    auto *advance_call =
+        codegen->CallBuiltin(ast::Builtin::IndexIteratorAdvance, {codegen->AddressOf(pg_statistic_index_iterator_)});
+
+    // for (@indexIteratorScanKey(&pg_statistic_index_iterator); @indexIteratorAdvance(&pg_statistic_index_iterator); )
+    Loop loop(function, loop_init, advance_call, nullptr);
+    {
+      // var pg_statistic_slot = @indexIteratorGetSlot(&pg_statistic_index_iterator)
+      auto pg_statistic_slot = codegen->MakeFreshIdentifier("pg_statistic_slot");
```

Comment:
@pmenon I was experiencing a really weird issue with this. I first tried to use the same identifier for each loop iteration (I mean C++ loop iteration not TPL), but my query would keep failing. The specific failure is that on the second table delete the slot value would be identical to the slot value from the first table delete. Since it was the same slot the delete was failing. Specifically `!Visible` was returning true here: https://github.com/cmu-db/noisepage/blob/d60c55a979e200500ad11b50661ad338be2007d6/src/storage/data_table.cpp#L209
because `not_deleted` was returning false here:
https://github.com/cmu-db/noisepage/blob/d60c55a979e200500ad11b50661ad338be2007d6/src/storage/data_table.cpp#L297-L301

I was able to fix it by just declaring a fresh identifier each loop iteration. Looking at the generated TPL the only difference I was able to see was the name of the slot variable. I was never able to figure out why this happened because the slot variable is declared in the TPL loop so it should be a new scope every time, even if it's the same variable name. 

I don't think it's a big issue because it's working now, I just wanted to point it out in case I'm misunderstanding something fundamental here.